File: sdk/python/pe/resources/validate.py
import httpx
from typing import Optional, List


class ValidateClient:
    def __init__(self, http: httpx.Client):
        self._http = http

    def validate(self, resource: dict, *, profile: Optional[str] = None,
                 enrich: bool = False, mode: str = "auto") -> dict:
        body = {"resource": resource, "enrich": enrich, "mode": mode}
        if profile:
            body["profile"] = profile
        return self._http.post("/validate", json=body).json()

    def fix(self, resource: dict, errors: Optional[List[dict]] = None) -> dict:
        body: dict = {"resource": resource}
        if errors is not None:
            body["errors"] = errors
        return self._http.post("/validate/fix", json=body).json()

    def profiles(self) -> dict:
        return self._http.get("/validate/profiles").json()

File: packages/sdk/src/resources/validate.ts
import type { HttpClient } from '../http';
import type { ValidationResult, FixResult, ValidateProfile } from '../types';

export interface ValidateOptions {
  profile?: string;
  enrich?: boolean;
  mode?: 'auto' | 'structural' | 'hl7';
}

export class ValidateClient {
  constructor(private readonly http: HttpClient) {}

  validate(resource: Record<string, unknown>, opts: ValidateOptions = {}): Promise<ValidationResult> {
    return this.http.post<ValidationResult>('/validate', { resource, ...opts });
  }

  fix(resource: Record<string, unknown>, errors?: unknown[]): Promise<FixResult> {
    return this.http.post<FixResult>('/validate/fix', { resource, ...(errors ? { errors } : {}) });
  }

  profiles(): Promise<{ profiles: ValidateProfile[] }> {
    return this.http.get('/validate/profiles');
  }
}
